File: 6KYU/Esolang Interpreters #1 - Introduction to Esolangs and My First Interpreter (MiniStringFuck)/solution.cs
public class Kata
{
  public static string MyFirstInterpreter(string code)
  {
     var res = "";
     var memory = 0;
     foreach(var c in code)
     {
        if(c == '+')
        {
          memory ++;
          if(memory == 256)
            memory = 0;
        }
        else if (c == '.')
        {
           res += ((char)memory).ToString();
        }  
     }
    
     return res;
  }
}

File: 6KYU/Esolang Interpreters #1 - Introduction to Esolangs and My First Interpreter (MiniStringFuck)/solution.go
package kata

func Interpreter(code string) string {
	memory := byte(0)
	res := []byte{}
	for _, c := range code {
		if c == '+' {
			if memory == 255 {
				memory = 0
			} else {
				memory += 1
			}
		} else if c == '.' {
			res = append(res, memory)
		}
	}

	return string(res[:])
}
